internal/postgres/data_test.go

```diff
@@ -0,0 +1,237 @@
+			pgTableDef{map[string]pgColDef{col: pgColDef{id: tc.pgt}}})
+		loc, _ := time.LoadLocation("Australia/Sydney")
+		conv.SetLocation(loc)
+		ac, av, err := ConvertData(conv, table, table, []string{col}, []string{tc.in})
+		assert.Nil(t, err, tc.name)
+		assert.Equal(t, []string{col}, ac, tc.name+": column mismatch")
+		// Avoid assert.Equal for time.Time (it forces location equality).
+		// Instead use Time.Equals, which determines equality based on whether
+		// two times represent the same instant.
+		assert.Equal(t, 1, len(av))
+		at, ok1 := av[0].(time.Time)
+		et, ok2 := tc.e.(time.Time)
+		assert.True(t, ok1 && ok2, tc.name+": cast to Time failed")
+		assert.True(t, at.Equal(et), tc.name+": value mismatch")
+	}
+
+	multiColTests := []struct {
+		name  string
+		cols  []string      // Input columns.
+		vals  []string      // Input values.
+		ecols []string      // Expected columns.
+		evals []interface{} // Expected values.
+	}{
+		{
+			name:  "Cols in order",
+			cols:  []string{"a", "b", "c"},
+			vals:  []string{"6", "6.6", "true"},
+			ecols: []string{"a", "b", "c"},
+			evals: []interface{}{int64(6), float64(6.6), true},
+		},
+		{
+			name:  "Cols out of order",
+			cols:  []string{"b", "c", "a"},
+			vals:  []string{"6.6", "true", "6"},
+			ecols: []string{"b", "c", "a"},
+			evals: []interface{}{float64(6.6), true, int64(6)},
+		},
+		{
+			name:  "Null column",
+			cols:  []string{"a", "b", "c"},
+			vals:  []string{"6", "\\N", "true"},
+			ecols: []string{"a", "c"},
+			evals: []interface{}{int64(6), true},
+		},
+		{
+			name:  "Missing columns",
+			cols:  []string{"a"},
+			vals:  []string{"6"},
+			ecols: []string{"a"},
+			evals: []interface{}{int64(6)},
+		},
+	}
+	spSchema := ddl.CreateTable{
+		"testtable",
+		[]string{"a", "b", "c"},
+		map[string]ddl.ColumnDef{
+			"a": ddl.ColumnDef{Name: "a", T: ddl.Int64{}},
+			"b": ddl.ColumnDef{Name: "b", T: ddl.Float64{}},
+			"c": ddl.ColumnDef{Name: "c", T: ddl.Bool{}},
+		},
+		[]ddl.IndexKey{},
+		"",
+	}
+	pgSchema := pgTableDef{
+		map[string]pgColDef{
+			"a": pgColDef{id: "int8"},
+			"b": pgColDef{id: "float8"},
+			"c": pgColDef{id: "bool"},
+		}}
+	for _, tc := range multiColTests {
+		conv := buildConv(spSchema.Name, spSchema, pgSchema)
+		acols, avals, err := ConvertData(conv, spSchema.Name, spSchema.Name, tc.cols, tc.vals)
+		checkResults(t, acols, avals, err, tc.ecols, tc.evals, tc.name)
+	}
+
+	errorTests := []struct {
+		name string
+		cols []string // Input columns.
+		vals []string // Input values.
+	}{
+		{
+			name: "Error in int64",
+			cols: []string{"a", "b", "c"},
+			vals: []string{" 6", "6.6", "true"},
+		},
+		{
+			name: "Error in float64",
+			cols: []string{"a", "b", "c"},
+			vals: []string{"6", "6.6e", "true"},
+		},
+		{
+			name: "Error in bool",
+			cols: []string{"a", "b", "c"},
+			vals: []string{" 6", "6.6", "truee"},
```

Comment:
'Error in bool' is returned because column 'c' is evaluated first? The reason I am asking is because column 'a' also has error in int64.

Reply:
Fixed.
Good catch -- thanks! Should be just a single error in the third arg (bool).